BSP/STM32F413/Src/BSP_CAN.c

```diff
@@ -1,144 +0,0 @@
-    CAN_InitStructure.CAN_Prescaler = 16;
-    CAN_Init(CAN1, &CAN_InitStructure);
-
-    /* CAN filter init */
-    CAN_FilterInitStructure.CAN_FilterNumber = 0;
-    CAN_FilterInitStructure.CAN_FilterMode = CAN_FilterMode_IdMask;
-    CAN_FilterInitStructure.CAN_FilterScale = CAN_FilterScale_32bit;
-    CAN_FilterInitStructure.CAN_FilterIdHigh = 0x0000;
-    CAN_FilterInitStructure.CAN_FilterIdLow = 0x0000;
-    CAN_FilterInitStructure.CAN_FilterMaskIdHigh = 0x0000;
-    CAN_FilterInitStructure.CAN_FilterMaskIdLow = 0x0000;
-    CAN_FilterInitStructure.CAN_FilterFIFOAssignment = 0;
-    CAN_FilterInitStructure.CAN_FilterActivation = ENABLE;
-    CAN_FilterInit(CAN1, &CAN_FilterInitStructure);
-
-    /* Transmit Structure preparation */
-    TxMessage.ExtId = 0x1;
-    TxMessage.RTR = CAN_RTR_DATA;
-    TxMessage.IDE = CAN_ID_STD;
-    TxMessage.DLC = 1;
-
-    /* Receive Structure preparation */
-    RxMessage.StdId = 0x00;
-    RxMessage.ExtId = 0x00;
-    RxMessage.IDE = CAN_ID_STD;
-    RxMessage.DLC = 0;
-    RxMessage.FMI = 0;
-
-    /* Enable FIFO 0 message pending Interrupt */
-    CAN_ITConfig(CAN1, CAN_IT_FMP0, ENABLE);
-
-    // Enable Rx interrupts
-    NVIC_InitStructure.NVIC_IRQChannel = CAN1_RX0_IRQn;
-    NVIC_InitStructure.NVIC_IRQChannelPreemptionPriority = 0x0;
-    NVIC_InitStructure.NVIC_IRQChannelSubPriority = 0x0;
-    NVIC_InitStructure.NVIC_IRQChannelCmd = ENABLE;
-    NVIC_Init(&NVIC_InitStructure);	
-}
-
-/**
- * @brief   Transmits the data onto the CAN bus with the specified id
- * @param   id : Message of ID. Also indicates the priority of message. The lower the value, the higher the priority.
- * @param   Voltage : the voltage value at that moment in a float
- * @return  0 if module was unable to transmit the data onto the CAN bus. Any other value indicates data was transmitted.
- */
-uint8_t BSP_CAN_Write(uint32_t id, float Voltage) {
-    TxMessage.StdId = id;
-    TxMessage.StdId = Voltage;
-	
-	return CAN_Transmit(CAN1, &TxMessage);
-}
-
-/**
- * @brief   Gets the data that was received from the CAN bus.
- * @note    Non-blocking statement
- * @pre     The data parameter must be at least 8 bytes or hardfault may occur.
- * @param   id : pointer to store id of the message that was received.
- * @param   data : pointer to store data that was received. Must be 8bytes or bigger.
- * @return  0 if nothing was received so ignore id and data that was received. Any other value indicates data was received and stored.
- */
-uint8_t BSP_CAN_Read(uint32_t *id, uint8_t *data) {
-    if(RxFlag){
-		for(int i = 0; i < 8; i++){
-			data[i] = RxMessage.Data[i];
-		}
-        *id = RxMessage.StdId;
-		RxFlag = false;
-		return 1;
-	}
-    return 0;
-}
-
-void CAN1_RX0_IRQHandler(void)
-{
-    CAN_Receive(CAN1, CAN_FIFO0, &RxMessage);
-
-    if ((RxMessage.StdId == 0x001)&&(RxMessage.IDE == CAN_ID_STD) && (RxMessage.DLC == 1)){
-        // TODO: do stuff
-        RxFlag = true;
-    }
-}
```

Comment:
Don't delete this file. You should edit this file to make any changes to the CAN BSP